cmd/libsnap-confine-private/privs.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "privs.h"
+
+#define _GNU_SOURCE
+
+#include <unistd.h>
+
+#include <grp.h>
+#include <stdbool.h>
+#include <sys/capability.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include "utils.h"
+
+static bool sc_has_capability(const char *cap_name)
+{
+	// Lookup capability with the given name.
+	cap_value_t cap;
+	if (cap_from_name(cap_name, &cap) < 0) {
```

Comment:
You're right but I thought as a library it might be handy to take as a string ('cap_chown' looks nice as a security policy writer after all :). It also returns -1 for unknown so we didn't need to do CAP_IS_SUPPORTED which if we didn't have this, we'd want to add.